relayer/crates/starknet-integration-tests/src/tests/clear_packets.rs

```diff
@@ -542,6 +543,26 @@ fn test_query_unreceived_packets() -> Result<(), Error> {
 
         runtime.sleep(Duration::from_secs(2)).await;
 
+        // approve ics20 contract to spend the tokens for `address_starknet_b`
+        {
+            let call_data = cairo_encoding.encode(&product![
+                ics20_contract_address,
+                U256::from(transfer_quantity)
+            ])?;
+
+            let call = Call {
+                to: *ics20_token_address,
+                selector: selector!("approve"),
+                calldata: call_data,
+            };
+
+            let execution = starknet_account_b.execute_v3(vec![call]);
+
+            let tx_hash = execution.send().await?.transaction_hash;
+
+            starknet_chain.poll_tx_response(&tx_hash).await?;
+        }
```

Comment:
hmm. this is a very interesting ! So, the token owner can burn tokens from anyone's balance, but token owner can't transfer it to their own balance ?

Reply:
Good question. This got me yesterday too. Should add a check in the `ERC20Mintable` contract.

In the standard ERC-20, since the supply is fixed, minting and burning don’t matter, so ownership isn’t really a concern. But in our case, we’re working with an extended version of ERC20 where only the owner (ICS20) can mint and burn tokens. This part of code was originally written at the start of the project just to get the MVP working. Now that it looks like we have no choice but to go with our own ERC20 version, it's good to review the implementation there and ensure it’s fully secure by adding some unit tests.